griptape/drivers/vector/marqo_vector_store_driver.py

```diff
@@ -0,0 +1,253 @@
+from typing import Optional, List, Dict, Any
+from griptape import utils
+from griptape.drivers import BaseVectorStoreDriver
+from griptape.artifacts import TextArtifact
+import marqo
+from attr import define, field
+import json
+
+
+@define
+class MarqoVectorStoreDriver(BaseVectorStoreDriver):
+    api_key: str = field(kw_only=True)
+    url: str = field(kw_only=True)
+    mq: marqo.Client = field(kw_only=True)
+    index: str = field(kw_only=True)
+
+    def __attrs_post_init__(self):
+        """Initialize the Marqo client with the given API key and URL."""
+        self.mq = marqo.Client(self.url, self.api_key)
```

Comment:
Could we set this one in the attribute factory? (here is an [example](https://github.com/griptape-ai/griptape/blob/06b119181f53e7fa34aabd597a8a9a4129712cd5/griptape/drivers/sql/amazon_redshift_sql_driver.py#L17))

Reply:
fixed :)